internal/app/machined/pkg/runtime/v1alpha1/bootloader/sdboot/sdboot.go

```diff
@@ -77,16 +77,18 @@ func ProbeWithCallback(disk string, options options.ProbeOptions, callback func(
 
 	// here we need to read the EFI vars to see if we have any defaults
 	// and populate config accordingly
-	// https://www.freedesktop.org/software/systemd/man/systemd-boot.html#LoaderEntryDefault
-	// this should be set on install/upgrades
+	// https://www.freedesktop.org/software/systemd/man/latest/systemd-boot.html#LoaderEntryDefault
+	// this is set by systemd-boot.
 	efiCtx := efivario.NewDefaultContext()
 
-	bootedEntry, err := ReadVariable(efiCtx, LoaderEntrySelectedName)
+	// first we start by checking if we have a Default entry
+	// this is set by installer
+	bootedEntry, err := ReadVariable(efiCtx, LoaderEntryDefaultName)
```

Comment:
we don't need to rely on the selected entry anymore since the condition below would pick the default uki if there's only one, allowing booting from iso, uki and initrd/kernel combo